File: JAVA/921. Minimum Add to Make Parentheses Valid.java
class Solution {
    public int minAddToMakeValid(String s) {      
        Stack<Character> st = new Stack();         
        for(int i=0;i<s.length();i++)
        {
            boolean bol = true;
            if(s.charAt(i)=='(')
            {
                 st.push('(');
                 bol = false;
            }
            else if(!st.empty() && s.charAt(i)==')' && st.peek()=='(')
            {
                st.pop();
                bol = false;
            }
            if(bol)
            st.push(s.charAt(i));
        }
        return st.size();
    }
}

File: C++/921. Minimum Add to Make Parentheses Valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        int balance = 0;
        int additions = 0;
        for (char c : s) {
            if (c == '(') {
                balance++;
            } else {
                balance--;
                if (balance < 0) {
                    additions++;
                    balance = 0;
                }
            }
        }
        return additions + balance;
    }
};
